lib/src/Base/IterativeStat/IterativeMoments.cxx

```diff
@@ -0,0 +1,294 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief IterativeMoments implements the iterative computation of centered moments
+ *
+ *  Copyright 2005-2021 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+#include "openturns/IterativeAlgorithm.hxx"
+#include "openturns/Log.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/Point.hxx"
+#include "openturns/IterativeMoments.hxx"
+#include "openturns/SpecFunc.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(IterativeMoments)
+
+static const Factory<IterativeMoments> Factory_IterativeMoments;
+
+/** Default constructor */
+IterativeMoments::IterativeMoments(const UnsignedInteger orderMax, const UnsignedInteger dimension)
+  : IterativeAlgorithmImplementation(dimension),
+    orderMax_(orderMax),
+    centeredMoments_(orderMax, dimension_)
+{
+  // Nothing to do
+}
+
+/** Virtual constructor */
+IterativeMoments * IterativeMoments::clone() const
+{
+  return new IterativeMoments(*this);
+}
+
+/** String converter */
+String IterativeMoments::__repr__() const
+{
+  OSS oss(true);
+  oss << "class=" << IterativeMoments::GetClassName()
+      << " iteration=" << iteration_
+      << " dimension=" << dimension_
+      << " order max=" << orderMax_
+      << " centered moments=" << centeredMoments_.__repr__();
+  return oss;
+}
+
+/** String converter */
+String IterativeMoments::__str__(const String & offset) const
+{
+  return getCenteredMoments().__str__(offset);
+}
+
+/**  Accessor to the centered moments */
+Sample IterativeMoments::getCenteredMoments() const
+{
+  return centeredMoments_;
+}
+
+/** Accessor to the maximum order declared */
+UnsignedInteger IterativeMoments::getOrder() const
+{
+  return orderMax_;
+}
+
+/** Accessor to the mean */
+Point IterativeMoments::getMean() const
+{
+  if (iteration_ >= 1)
+  {
+    return centeredMoments_[0] / iteration_;
+  }
+  else
+  {
+    return centeredMoments_[0];
```

Comment:
You should throw an exception if iteration_ == 0, see SampleImplementation::computeMean()

Reply:
There is no mean if the sample is empty and an exception should be raised, as mentioned by R.L.. The getMomentOne method does not seem to fix that particular issue.